services/renderFabricFrame.js

```diff
@@ -0,0 +1,91 @@
+import { fabric } from "fabric";
+import fixRenderCoord from "../utils/fixRenderCoord";
+import typeMapper from "./createFabricObjects";
+
+const matchType = async (el, number) => {
+  if (!el.type) {
+    const getDefaultFunction = typeMapper.RECTANGLE;
+    const fabricObject = await getDefaultFunction(el, number);
+
+    return fabricObject;
+  }
+
+  const getTypeFunction = typeMapper[el.type];
+
+  if (getTypeFunction) {
+    const fabricObject = await getTypeFunction(el, number);
+
+    return fabricObject;
+  }
+};
+
+const renderFabricFrame = async function (frameJSON, imageUrl) {
+  const fabricObject = new Map();
+
+  if (frameJSON.type === "NEW_FRAME") {
+    frameJSON.type = "";
+  }
+
+  const parent = await matchType(frameJSON, fixRenderCoord(frameJSON));
+
+  parent.stroke = "black";
+  parent.strokeWidth = 2;
+
+  fabricObject.set(frameJSON.frameId, parent);
+
+  for (const nodeId in frameJSON.nodes) {
+    if (frameJSON.nodes[nodeId].property.fills[0]?.imageRef) {
+      const { imageRef } = frameJSON.nodes[nodeId].property.fills[0];
+
+      if (imageUrl[imageRef]) {
+        frameJSON.nodes[nodeId].property.imageURL = imageUrl[imageRef];
+      }
+    }
```

Comment:
- `P2`
```suggestion
  for (const nodeId in frameJSON.nodes) {
    if (frameJSON.nodes.hasOwnProperty(nodeId)) {
        if (frameJSON.nodes[nodeId].property.fills[0]?.imageRef) {
        const { imageRef } = frameJSON.nodes[nodeId].property.fills[0];

        if (imageUrl[imageRef]) {
          frameJSON.nodes[nodeId].property.imageURL = imageUrl[imageRef];
        }
      }
    }
```

`hasOwnProperty`를 작성해주시면 `frameJSON.nodes`가 온전히 갖고있는 property만 순회가 가능 할 것 같습니다!
예상치 못한 사이드 이펙트 방지에 도움이 될 것 같아요

Reply:
해당 에러는 `Object.prototype.hasOwnProperty.call(object, targetProprety)` 로 작성하시면 해결 됩니다!